src/controllers/organizationController.ts

```diff
@@ -317,204 +244,339 @@ export default class OrganizationController {
     });
   }
 
-  public async sendDocuments(
-    req: Request
-  ): Promise<
-    // tslint:disable-next-line:max-union-size
-    | IResponseErrorValidation
-    | IResponseErrorForbiddenNotAuthorized
-    | IResponseErrorNotFound
-    | IResponseErrorConflict
-    | IResponseErrorInternal
-    | IResponseNoContent
-  > {
-    return withUserFromRequest(req, async user => {
-      const userPermissions = accessControl.can(user.role);
-      if (!userPermissions.createOwn(Resource.SIGNED_DOCUMENT).granted) {
-        return ResponseErrorForbiddenNotAuthorized;
-      }
-      const errorOrMaybeOrganizationInstance = await getOrganizationInstanceFromDelegateEmail(
-        user.email,
-        req.params.ipaCode
-      );
-      return errorOrMaybeOrganizationInstance.fold(
-        async error => {
-          log.error("An error occurred reading from db. %s", error);
-          return ResponseErrorInternal("An error occurred reading from db");
-        },
-        async maybeOrganizationInstance => {
-          if (isNone(maybeOrganizationInstance)) {
-            return ResponseErrorNotFound(
-              "Not found",
-              "The administration is not registered"
-            );
-          }
-          const organizationInstance = maybeOrganizationInstance.value;
-          if (
-            organizationInstance.registrationStatus ===
-            OrganizationRegistrationStatusEnum.REGISTERED
-          ) {
-            return ResponseErrorConflict(
-              "The required documents have already been sent and countersigned"
-            );
-          }
-          return this.signAndSendDocuments(
-            user.fiscalCode,
-            organizationInstance
-          );
-        }
+  /**
+   * Submits an onboarding request by sending to the related administration an email containing the requested signed documents.
+   * In order to achieve the result, this method performs several steps:
+   * 1. Fetches the user from the request
+   * 2. Checks the user permissions
+   * 3. Validates the body request
+   * 4. Fetches and validates the requests to be submitted
+   * 5. Creates the requested signed documents
+   * 6. Sends the documents via email
+   * 7. Updates the status of the requests
+   *
+   * If any error occurs during any step of the process, a proper error response is sent to the client; otherwise, a response with no content is sent.
+   * @param req The client request
+   */
+  public sendDocuments(
```

Comment:
a more explicit method name here would be nice, ie.

```suggestion
  public sendEmailWithDocumentsToOrganizationPec(
```